### ISSUE TITLE
Impossible to emerge gazebo (noetic) packages because gazebo won't emerge

### ISSUE THREAD

**Issue by flabrosse** [ros/ros-overlay#1009]
Trying to emerge noetic gazebo packages and this fails because I can't emerge sci-electronics/gazebo.  The latest in the tree fails because of linking issues with openal, older versions also fail due to missing older dependencies.

The problem is with [openal](https://bugs.gentoo.org/734806).  Nothing seems to have been done about it.  Is it worth making a package here that includes the patch?  @allenh1 ?

**Comment by allenh1** [comment]
@flabrosse if it would help you, it's certainly worth doing.

I've been very swamped lately, but would help you in any way I could to get the package in. Otherwise, it could honestly be a while until I get the time to fix things around here.

**Comment by flabrosse** [comment]
Thanks @allenh1.  I've been swamped too, but did my last lecture today so have a bit of time now.  I created a few PRs on other aspects and noticed that an old one (not from me) seems to solve many problems (including one I covered in one my PRs).  I'll work on the gazebo one next week now.

**Comment by allenh1** [comment]
I'm take a look at those as soon as I can, and try and get them merged in. Please @-me on any that are important!

**Comment by flabrosse** [comment]
Obviously, making a ros-noetic/gentoo package might will create conflicts with the gentoo in-tree version if installed on it's own or via other dependencies.  I have no idea how to solve this problem...  I also don't know how to look for other packages that might require it.

**Comment by focusrobotics** [comment]
I hit this too and found https://bugs.gentoo.org/734806 which addresses it and provides a patch. Gazebo 11.3.0 compiled successfully with that patch applied.

**Comment by allenh1** [comment]
>  Gazebo 11.3.0 compiled successfully with that patch applied.

Great!

All, I'd like to apologize for my lack of attention being paid here. I'll try to be more active.

**Comment by flabrosse** [comment]
No worries.  We are all busy.

Since a few days ago there was an update of boost, and since then gazebo has failed to emerge, linking errors with openal.  I have reemerged openal, and many other packages (in fact the whole dependency tree of gazebo), to no avail.  I think it's because there is a package that is not marked as an explicit dependency that needs recompiling but because the link is not there portage does not know about it.

To test my hypothesis I'll re-emerge @system and @world (will use less of my time than trying to manually track down the missing dependency...).

**Comment by flabrosse** [comment]
I have re-emerged everything on my computer and gazebo still fails with linking issues with some openal functions.  I have no idea what is going on.  Any suggestion would be greatly welcomed.

**Comment by focusrobotics** [comment]
I tried re-emerging gazebo but got no errors. I synced and updated my system just a few days ago and definitely saw that boost updated.

Are you sure you are properly applying the patch from https://bugs.gentoo.org/734806 ?
If you are, can you post some more details on exactly what error you are getting?

**Comment by flabrosse** [comment]
Arghhhh.  I was, and then my copy got overridden by an update, without me noticing.  Thanks @focusrobotics.

**Comment by focusrobotics** [comment]
Yeah, I've done that before. You can also put the diff in a file called /etc/portage/patches/sci-electronics/gazebo/openal_link.patch and it will be automatically applied and won't be overwritten when you sync forward again. You can tell it was applied because it prints "User patches applied" during the merge.

**Comment by flabrosse** [comment]
I did not know about this feature.  I've always added my patches either in the main tree (and then losing them) or in an overlay tree with a bumped up r number.  Thanks for the trick!  And now gazebo emerges...
